### ISSUE TITLE
[Bug?]: JS styleObjects should use `dataDisabled`

### ISSUE THREAD

**Issue by caseybaggz** [pluralsight/pando#1001]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Currently, the generated JS styles are returning the CSS syntax `data-disabled` which causes this error logged when using React:

```bash
Warning: Unsupported style property &:active:not([data-disabled='true']). Did you mean &:active:not([dataDisabled='true'])?
```

This issue also relates to Discussion #741 

### Expected behavior 🤔

At the base level, the generated files need to return the correct `dataDisabled` syntax. However, this might require an [RFC to be created](https://github.com/pluralsight/pando-rfcs) for potentially designing a refactor that ultimately solves the issue via the Discussion topic mentioned above.

### Steps to reproduce 🕹

Steps:

1. Open a sandbox using the latest next build of HS
2. use the `getJSButtonProps` helper
3. LOOK AT CONSOLE LOG to view error
4.

**Comment by caseybaggz** [comment]
Also getting this on:

```bash
Warning: Unsupported style property &[data-invalid="true"]:hover. Did you mean &[dataInvalid="true"]:hover?
```

**Comment by josh-degraw** [comment]
@caseybaggz @Rykus0 
I just saw a PR that addressed this just got merged, and I think that it should be reverted. I had filed an [issue](https://github.com/pluralsight/pando/issues/818) related to this console output a while ago and from looking into that I realized what was happening.

The reason that the warning is getting logged by react is because it thinks you're trying to use e.g. `&[data-invalid="true"]:hover` as an untranslated CSS property.

If you try to pass this to a plain react component, it will have no idea what to do with it, hence the warning. See this [codesandbox](https://codesandbox.io/s/smoosh-cdn-c8jcgs?file=/src/App.js) with the console error shown. I believe the reason it's suggesting to use camelcased is because react needs all style properties to be, e.g. `padding-top` becomes `paddingTop` because the former isn't valid JS.

In short, the problem here isn't the casing, (which should remain using valid CSS selectors in sub-selectors), but the fact that we (myself included in the past) were incorrectly attempting to pass these style objects directly to the JSX `style` prop. That won't work, and never would have, because the `&`-prefixed nested selector syntax is not valid for inline styles. However, other libraries such as Chakra & MUI support it in various ways (e.g. we can consume them directly in MUI even inline via the [sx prop](https://mui.com/material-ui/react-box/#the-sx-prop) or in theme configuration). But for them to work as expected they need to be using proper CSS pseudo selectors in nested/composed selectors (e.g. anything with `&`). If I tried to use the style objects with the camelcased pseudo selectors, it wouldn't work anymore.
